### PR TITLE
fix: fix onlineddl parse keyspace

### PR DESCRIPTION
**Pull request by gerayking**
## Description
fix onlineddl keyspace and dbName

## Checklist

-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/schema/online_ddl.go [hunk 1]
```diff
@@ -214,9 +214,10 @@ func NewOnlineDDL(keyspace string, table string, sql string, ddlStrategySetting
 			return strconv.Quote(hex.EncodeToString([]byte(directive)))
 		}
 		comments := sqlparser.Comments{
-			fmt.Sprintf(`/*vt+ uuid=%s context=%s table=%s strategy=%s options=%s */`,
+			fmt.Sprintf(`/*vt+ uuid=%s context=%s schema=%s table=%s strategy=%s options=%s */`,
```
Comment (earayu) [inline go/vt/schema/online_ddl.go:217]: `tableSchema` would be better

File: go/vt/schema/online_ddl.go [hunk 2]
```diff
@@ -276,8 +277,10 @@ func formatWithoutComments(buf *sqlparser.TrackedBuffer, node sqlparser.SQLNode)
 // to be commented as e.g. `CREATE /*vt+ uuid=... context=... table=... strategy=... options=... */ TABLE ...`
 func OnlineDDLFromCommentedStatement(stmt sqlparser.Statement) (onlineDDL *OnlineDDL, err error) {
 	var comments *sqlparser.ParsedComments
+	schemaName := ""
 	switch stmt := stmt.(type) {
```
Comment (earayu) [inline go/vt/schema/online_ddl.go:281]: tableSchema

File: go/vt/schema/online_ddl.go [hunk 3]
```diff
@@ -306,14 +309,18 @@ func OnlineDDLFromCommentedStatement(stmt sqlparser.Statement) (onlineDDL *Onlin
...
 	}
 	if onlineDDL.UUID, err = decodeDirective("uuid"); err != nil {
 		return nil, err
 	}
 	if !IsOnlineDDLUUID(onlineDDL.UUID) {
 		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "invalid UUID read from statement %s", sqlparser.String(stmt))
 	}
+	if schemaName, err = decodeDirective("schema"); err != nil && onlineDDL.Keyspace == "" {
+		onlineDDL.Keyspace = schemaName
+	}
```
Comment (earayu) [inline go/vt/schema/online_ddl.go:323]: should `onlineDDL.Schema` also be assigned?

File: go/vt/vttablet/onlineddl/executor.go [hunk 4]
```diff
@@ -4709,9 +4714,9 @@ func (e *Executor) SubmitMigration(
 	_, allowConcurrentMigration := e.allowConcurrentMigration(onlineDDL)
 	submitQuery, err := sqlparser.ParseAndBind(sqlInsertMigration,
 		sqltypes.StringBindVariable(onlineDDL.UUID),
-		sqltypes.StringBindVariable(e.keyspace),
+		sqltypes.StringBindVariable(onlineDDL.Keyspace),
 		sqltypes.StringBindVariable(e.shard),
```
Comment (earayu) [inline go/vt/vttablet/onlineddl/executor.go:4718]: `onlineDDL.Schema`

File: go/vt/schema/online_ddl.go (hunk 3 above)
Comment (gerayking) [reply go/vt/schema/online_ddl.go:323]: Yes, the `onlineDDL.Schema` should also be assigned. However, I believe we should assign either `Keyspace` or `Schema`, but not both.

**Comment by gerayking** [comment]
see #278

**Pull request closed without merging**